app/jobs/whats_app_adapter/three_sixty_dialog/create_templates_job.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require 'net/http'
+
+module WhatsAppAdapter
+  module ThreeSixtyDialog
+    class CreateTemplatesJob < ApplicationJob
+      def perform(organization_id:)
+        @organization = Organization.find(organization_id)
+
+        @base_uri = ENV.fetch('THREE_SIXTY_DIALOG_WHATS_APP_REST_API_ENDPOINT', 'https://stoplight.io/mocks/360dialog/360dialog-partner-api/24588693')
+        existing_templates = WhatsAppAdapter::ThreeSixtyDialog::TemplateFetcherService.new(organization_id: organization.id).call
+
+        templates_to_create_array = whats_app_templates.keys.difference(existing_templates)
+        templates_to_create = whats_app_templates.select { |key, _value| key.in?(templates_to_create_array) }
+        templates_to_create.each do |key, value|
+          @template_name = key
+          @template_text = value
```

Comment:
This looks weird, if we're in a loop, these two instance variables get overwritten on each iteration. Only the last iteration will matter.